tests/mpi/solution_transfer_11.cc

```diff
@@ -0,0 +1,138 @@
+#include <deal.II/dofs/dof_tools.h>
+
+#include <deal.II/fe/fe_q.h>
+
+#include <deal.II/grid/grid_generator.h>
+
+#include <deal.II/hp/fe_collection.h>
+
+#include <deal.II/lac/la_parallel_vector.h>
+
+#include "../tests.h"
+
+
+template <int dim>
+void
+test()
+{
+  parallel::distributed::Triangulation<dim> tria(MPI_COMM_WORLD);
+  GridGenerator::subdivided_hyper_cube(tria, 2);
+  tria.refine_global(1);
+
+  hp::FECollection<dim> fes;
+  for (unsigned int d = 1; d <= 3; ++d)
+    fes.push_back(FE_Q<dim>(d));
+
+  DoFHandler<dim> dofh(tria);
+  for (const auto &cell : dofh.active_cell_iterators())
+    if (cell->is_locally_owned())
+      cell->set_active_fe_index(1);
+
+  dofh.distribute_dofs(fes);
+  IndexSet locally_owned_dofs = dofh.locally_owned_dofs();
+  IndexSet locally_relevant_dofs =
+    DoFTools::extract_locally_relevant_dofs(dofh);
+
+  // set up solution
+  LinearAlgebra::distributed::Vector<double> solution;
+  solution.reinit(locally_owned_dofs,
+                  locally_relevant_dofs,
+                  dofh.get_communicator());
+
+  for (unsigned int i = 0; i < solution.size(); ++i)
+    if (locally_owned_dofs.is_element(i))
+      solution(i) = i;
+  solution.update_ghost_values();
+
+  // set refine/coarsen flags manually
+  for (const auto &cell : dofh.active_cell_iterators())
+    if (cell->is_locally_owned())
+      {
+        const std::string parent_id_string = cell->parent()->id().to_string();
+
+        if (parent_id_string == "0_0:")
+          {
+            // parent 0: h refinement
+            cell->set_refine_flag();
+          }
+        else if (parent_id_string == "1_0:")
+          {
+            // parent 1: h coarsening
+            cell->set_coarsen_flag();
+          }
+        else if (parent_id_string == "2_0:")
+          {
+            // parent 2: p refinement
+            const auto super_fe_index =
+              fes.next_in_hierarchy(cell->active_fe_index());
+            cell->set_future_fe_index(super_fe_index);
+          }
+        else if (parent_id_string == "3_0:")
+          {
+            // parent 3: p coarsening
+            const auto sub_fe_index =
+              fes.previous_in_hierarchy(cell->active_fe_index());
+            cell->set_future_fe_index(sub_fe_index);
+          }
+        else
+          {
+            Assert(false, ExcInternalError());
+          }
+      }
+
+  // initiate refinement and transfer
+  parallel::distributed::
+    SolutionTransfer<dim, LinearAlgebra::distributed::Vector<double>>
+      soltrans(dofh);
+  soltrans.prepare_for_coarsening_and_refinement(solution);
+
+  tria.execute_coarsening_and_refinement();
+
+  dofh.distribute_dofs(fes);
+  locally_owned_dofs    = dofh.locally_owned_dofs();
+  locally_relevant_dofs = DoFTools::extract_locally_relevant_dofs(dofh);
+
+  solution.reinit(locally_owned_dofs,
+                  locally_relevant_dofs,
+                  dofh.get_communicator());
+  soltrans.interpolate(solution);
```

Comment:
This is the end of the test. Would it make sense to compute some diagnostic information from the so-interpolated vector? Say its l1 norm or some something one could output? Right now, all you test is that the test doesn't abort.

Reply:
I've added the output of the l1 norm pre and post refinement.

Reply:
Great, thanks!